modules/face/include/opencv2/face/face_alignment.hpp

```diff
@@ -0,0 +1,84 @@
+    without specific prior written permission.
+
+This software is provided by the copyright holders and contributors "as is" and
+any express or implied warranties, including, but not limited to, the implied
+warranties of merchantability and fitness for a particular purpose are
+disclaimed. In no event shall copyright holders or contributors be liable for
+any direct, indirect, incidental, special, exemplary, or consequential damages
+(including, but not limited to, procurement of substitute goods or services;
+loss of use, data, or profits; or business interruption) however caused
+and on any theory of liability, whether in contract, strict liability,
+or tort (including negligence or otherwise) arising in any way out of
+the use of this software, even if advised of the possibility of such damage.
+*/
+#ifndef __OPENCV_FACE_ALIGNMENT_HPP__
+
+#define __OPENCV_FACE_ALIGNMENT_HPP__
+
+#include "opencv2/core.hpp"
+
+namespace cv{
+namespace face{
+/** @brief Implementation of face alignnment from the paper:
+ *  <a href="https://pdfs.semanticscholar.org/d78b/6a5b0dcaa81b1faea5fb0000045a62513567.pdf">Vahid Kazemi and Josephine Sullivan, "One Millisecond Face Alignment with an Ensemble of Regression Trees."
+ *  Computer Vision and Pattern Recognition, 2014. CVPR 2014.</a>
+ */
+class CV_EXPORTS FacemarkKazemi
+{
+public:
+        /** @returns The number of landmarks detected in an image */
+        virtual int getNumLandmarks() const = 0;
+        /** @returns The number of faces detected in an image */
+        virtual int getNumFaces() const = 0;
+        /** Implemented in @ref FacemarkKazemiimpl
+        */
+        virtual bool getData(std::vector<String> filename,std::vector<std::vector<Point2f> > & trainlandmarks
+                ,std::vector<String> & trainimages)=0;
+        /** Implemented in @ref cv::face::FacemarkKazemiimpl
+        */
+        virtual bool getMeanShapeRelative(std::vector<Rect> face,std::vector< std::vector<Point2f> > & initialshape)=0;
+        /** Implemented in @ref cv::face::FacemarkKazemiimpl
+        */
+        virtual bool getFaces(Mat src,std::vector<Rect> &facep,int detector_select=1,double scaleFactor=1.05
+        , int minNeighbors=3, int flags=0, Size minSize=Size(30,30), Size maxSize=Size()) =0;
+        /** Implemented in @ref FacemarkKazemiimpl
+        */
+        virtual bool calcMeanShape(std::vector< std::vector<Point2f> > & trainlandmarks,InputArrayOfArrays trainimages)=0;
+        /** Implemented in @ref FacemarkKazemiimpl
+        */
+        virtual bool scaleData(std::vector< std::vector<Point2f> > & trainlandmarks,
+                                    std::vector<Mat> & trainimages , Size s=Size(200,200) )=0;
+        /** Implemented in @ref cv::face::FacemarkKazemiimpl
+        */
+        virtual bool train(std::vector<Mat> images, std::vector< std::vector<Point2f> > landmarks)=0;
+        virtual ~FacemarkKazemi()=0;
+};
+
+CV_EXPORTS Ptr<FacemarkKazemi> create();
+}//face
+}//cv
```

Comment:
In GENERAL for the header
- Each function should have a better explanation in docs than, implemented in ...
- Add a description for all input and output parameters and possible return values
- Also each function needs a small explanation of what is happening when calling it

For now, the documentation is lacking behind that standard, so try to improve this.